### PR TITLE
[EventHubs] add logging.info to warn the usage of partition key of non-string type

### PR DESCRIPTION
**Pull request by yunhaoling**
The PR is to further address the issue with stronger warning in logging: https://github.com/Azure/azure-sdk-for-python/issues/12353

### CODE REVIEW

**Review by swathipil — APPROVED** [review]
:shipit:

File: sdk/eventhub/azure-eventhub/azure/eventhub/_producer_client.py [hunk 1]
```diff
@@ -246,6 +246,12 @@ def send_batch(self, event_data_batch, **kwargs):
         """
         partition_id = kwargs.get("partition_id")
         partition_key = kwargs.get("partition_key")
+
+        if partition_key and not isinstance(partition_key, str):
+            _LOGGER.info("WARNING: Please DO NOT pass a partition_key of non-string type to the send_batch method."
```
Comment (rakshith91) [inline sdk/eventhub/azure-eventhub/azure/eventhub/_producer_client.py:251]: why not `_looger.warning`?

File: sdk/eventhub/azure-eventhub/azure/eventhub/_producer_client.py [hunk 2]
```diff
@@ -306,6 +312,11 @@ def create_batch(self, **kwargs):
         partition_id = kwargs.get("partition_id", None)
         partition_key = kwargs.get("partition_key", None)
 
+        if partition_key and not isinstance(partition_key, str):
+            _LOGGER.info("WARNING: Please DO NOT pass a partition_key of non-string type to the create_batch method."
```
Comment (rakshith91) [inline sdk/eventhub/azure-eventhub/azure/eventhub/_producer_client.py:316]: same here

File: sdk/eventhub/azure-eventhub/azure/eventhub/_producer_client.py (hunk 1 above)
Comment (rakshith91) [reply sdk/eventhub/azure-eventhub/azure/eventhub/_producer_client.py:251]: this applies everywhere - is there a specific reason why you restricted it to info than a warning

Comment (yunhaoling) [reply sdk/eventhub/azure-eventhub/azure/eventhub/_producer_client.py:251]: according to our guideline warning is used to indicate error and error should be raised:
> DO use the WARNING logging level when a function fails to perform its intended task. The function should also raise an exception.

while in special case passing partition_key of non-string type doesn't cause any error, the service simply ignore, so making it a logging info.

Comment (annatisch) [reply sdk/eventhub/azure-eventhub/azure/eventhub/_producer_client.py:251]: Yeah I think `info` is probably appropriate here - we can bump it up a level later if we get feedback, or messaging from the service team that this should be treated more like an error. Though currently the service is not treating this like an error.
Do we want to tell people _not_ to do this? Or be informative, e.g. "By settings a non-string value, this partition key will be ignored by the service"
(Just asking, not suggesting :) )

File: sdk/eventhub/azure-eventhub/azure/eventhub/_producer_client.py [hunk 3]
```diff
@@ -306,6 +312,11 @@ def create_batch(self, **kwargs):
         partition_id = kwargs.get("partition_id", None)
         partition_key = kwargs.get("partition_key", None)
 
+        if partition_key and not isinstance(partition_key, str):
```
Comment (annatisch) [inline sdk/eventhub/azure-eventhub/azure/eventhub/_producer_client.py:315]: Does this need to be `six.string_types`? Or would 2.7 not support a unicode value here?

File: sdk/eventhub/azure-eventhub/azure/eventhub/_producer_client.py (hunk 1 above)
Comment (yunhaoling) [reply sdk/eventhub/azure-eventhub/azure/eventhub/_producer_client.py:251]: based on:
- what Serkant mentioned in the issue #12353 :
> Service doesn't do any validation on the delivered type and assumes sender library already conducted sanitary checks. This applies to every entry in the annotations map as far as I know.
- the customer described having trouble with Azure Function receiving messages which uses SDKs from a different language trying to parse the value:
```Unable to cast object of type 'System.Int32' to type 'System.String```

I think users had better not to set the partition_key of non-string type.

However, the wording could be improved in a "softer" way, I'm thinking of this way to be less imperative and more descriptive.

`Setting a partition_key of non-string type is discouraged in which case the partition_key will be ignored by the Event Hub service ignores and events will be assigned to all partitions using round-robin. Furthermore, SDKs from other languages might fail to parse the non-string value if they only expect partition_key to be string type`

How does it look to you?

File: sdk/eventhub/azure-eventhub/azure/eventhub/_producer_client.py (hunk 3 above)
Comment (yunhaoling) [reply sdk/eventhub/azure-eventhub/azure/eventhub/_producer_client.py:315]: just tested unicode partition_key in python2.7, they are working well and can be distributed among different partitions (absolutely not in a round-robin way)

Comment (yunhaoling) [reply sdk/eventhub/azure-eventhub/azure/eventhub/_producer_client.py:315]: I think this should actually be `isinstance(partition_key, (six.text_type, six.binary_type))` depending on our uamqp util.data_factory implementation which is used to encode the `partiton_key` inside the message annotation, six.text_type and six.binary_type will be interpreted into AMQPString value.

I have also tested partition_key as bytes, they're working as well.

https://github.com/Azure/azure-uamqp-python/blob/e1051ae1c8f93ed2792a852857d261a4f34b76f5/uamqp/utils.py#L117-L120

```python
def data_factory(value, encoding='UTF-8'):
    ...
    elif isinstance(value, six.text_type):
        result = c_uamqp.string_value(value.encode(encoding))
    elif isinstance(value, six.binary_type):
        result = c_uamqp.string_value(value)
    ...
```

File: sdk/eventhub/azure-eventhub/azure/eventhub/_producer_client.py (hunk 1 above)
Comment (annatisch) [reply sdk/eventhub/azure-eventhub/azure/eventhub/_producer_client.py]: Looks good! Thanks!

**Pull request merged**